frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx

```diff
@@ -244,6 +233,39 @@ export const DataPickerModal = ({
       onClose={onClose}
       onItemSelect={handleItemSelect}
       isLoadingTabs={isLoadingAvailableData}
+      searchExtraButtons={[filterButton]}
     />
   );
 };
+
+const FilterButton = ({
+  value,
+  onChange,
+  options,
+}: {
+  value: CollectionItemModel[];
+  onChange: (value: CollectionItemModel[]) => void;
+  options: FilterOption[];
+}) => {
+  return (
+    <Popover zIndex={1000}>
+      <Popover.Target>
+        <Button leftIcon={<Icon name="filter" />} variant="subtle">{c(
+          "A verb, not a noun",
+        ).t`Filter`}</Button>
+      </Popover.Target>
+      <Popover.Dropdown>
+        <Checkbox.Group value={value} onChange={onChange} px="1rem" py="0.5rem">
+          {options.map(option => (
+            <Checkbox
+              key={`filter-${option.value}`}
+              label={option.label}
+              value={option.value}
+              my="1rem"
+            />
+          ))}
+        </Checkbox.Group>
+      </Popover.Dropdown>
+    </Popover>
```

Comment:
padding looks kinda funny to me

![Screen Shot 2024-12-03 at 12 43 14 PM](https://github.com/user-attachments/assets/458148af-6c61-4ea6-bf09-a5389b66273c)

Reply:
@iethree Are you referring to how the outline juts out? If so, I imagine it's supposed to do that, but I might be wrong